Proposal: Category tagging and id edits for documentation

awcrosby opened the pull request:
In the near future the ansible galaxy dev team would like to use the default ansible-lint rules (lib/ansiblelint/rules) as part of the galaxy import process. This PR is a proposal to organize the default rules to help with display, documentation, and adding future rules. We are looking for community feedback.

Proposal:
- Have each rule be in a category, and have the id relate to the category
- Use the first tag in the rule `tags` list to denote the category
- If that seems acceptable, I took a pass at editing the tags and ids, including shortening the ids. The result matches the [pep8 error code setup](https://pep8.readthedocs.io/en/latest/intro.html#error-codes), and enables auto-generated documentation: RULE_DOCS.md

willthames commented:
I am all in favour of this change, other than that I would like to retain a mapping of historic names to new IDs for backward compatibility purposes so that people using `-t`, `-x` or `skip_list` don't have to update their usage. 

I would be fine with a deprecation cycle of a couple of releases

tima commented:
Understand what you are saying @willthames. The way this PR is done, the output would change in that these new python codestyle IDs would display and not the historic ones being used today. Would this be too much of a break? Is using that ID when parsing results a common use case?

willthames commented:
@tima I'm not too worried about the display names as long as existing ansible-lint invocations work in the same way 

We do have a ParseableFormatter so that jenkins junit style reports work correctly - it's worth checking that these new names behave correctly with the ParseableFormatter - `ansible-lint -p`. Looking at the code, the ParseableFormatter prepends `E` to the rule id, so you'll end up with EE101 when using `-p`.

awcrosby commented:
I updated the PR based on this conversation:
- added the historic id as a tag for each rule
- added a test case to check running and skipping both by rule id and tag
- added example to README `skip_list`

willthames commented:
This can go into 3.5.0
